app/src/main/java/org/mozilla/fenix/tabstray/TrayPagerAdapter.kt

```diff
@@ -76,30 +70,38 @@ class TrayPagerAdapter(
             }
             PrivateBrowserPageViewHolder.LAYOUT_ID -> {
                 PrivateBrowserPageViewHolder(
-                    itemView,
+                    LayoutInflater.from(parent.context).inflate(viewType, parent, false),
                     tabsTrayStore,
                     browserStore,
                     interactor
                 )
             }
             SyncedTabsPageViewHolder.LAYOUT_ID -> {
                 SyncedTabsPageViewHolder(
-                    itemView,
-                    tabsTrayStore
+                    composeView = ComposeView(parent.context).apply {
+                        layoutParams = ViewGroup.LayoutParams(
+                            ViewGroup.LayoutParams.MATCH_PARENT,
+                            ViewGroup.LayoutParams.MATCH_PARENT
+                        )
+                    },
+                    tabsTrayStore = tabsTrayStore,
+                    navigationInteractor = navInteractor
                 )
             }
             else -> throw IllegalStateException("Unknown viewType.")
         }
-    }
 
+    /**
+     * Until [TrayPagerAdapter] is replaced with a Compose implementation, [SyncedTabsPageViewHolder]
+     * will need to be called with an empty bind() function since it no longer needs an adapter to render.
+     * For more details: https://github.com/mozilla-mobile/fenix/issues/21318
+     */
     override fun onBindViewHolder(viewHolder: AbstractPageViewHolder, position: Int) {
-        val adapter = when (position) {
-            POSITION_NORMAL_TABS -> normalAdapter
-            POSITION_PRIVATE_TABS -> privateAdapter
-            POSITION_SYNCED_TABS -> syncedTabsAdapter
-            else -> throw IllegalStateException("View type does not exist.")
+        when (viewHolder) {
+            is NormalBrowserPageViewHolder -> viewHolder.bind(normalAdapter)
+            is PrivateBrowserPageViewHolder -> viewHolder.bind(privateAdapter)
+            is SyncedTabsPageViewHolder -> viewHolder.bind()
```

Comment:
It might be worthwhile to align more closely with `SessionControlAdapter`. https://searchfox.org/mozilla-mobile/source/fenix/app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlAdapter.kt#361

Also wondering if we need to do something like https://searchfox.org/mozilla-mobile/source/fenix/app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlAdapter.kt#293.

Reply:
align with which part? not needing bind() call?

Reply:
Ya not needing to call bind(), I don't think we need to do this here. This can be a followup.

Reply:
Reading back through this, we also may not even need to file a separate follow-up, since the real follow-up is converting the rest of TabsTray to Compose